NeoOnnx/src/Optimizer.h

```diff
@@ -0,0 +1,75 @@
+/* Copyright © 2017-2020 ABBYY Production LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--------------------------------------------------------------------------------------------------------------*/
+
+#pragma once
+
+#include "common.h"
+#include <NeoOnnx/NeoOnnxImport.h>
+
+namespace NeoOnnx {
+
+class IOptimizer : public IObject {
+public:
+	explicit IOptimizer( CDnn& graph ) : graph( graph ) {}
+	IOptimizer( IOptimizer&& ) = delete;
+	IOptimizer( const IOptimizer& ) = delete;
+
+	virtual void Apply() = 0;
+
+protected:
+	CDnn& graph;
+	CArray<CPtr<CBaseLayer>> layersToRemove{};
+
+	static constexpr const char* const classesOfSkipLayers[]{
+		"NeoMLDnnBroadcastLayer",
+		"FmlCnnTransformWithoutTransposeLayer",
+		"FmlCnnTransposeLayer"
+	};
+
+	/// Returns the pointer to input layer of the 'currentLayer' of any class-type, but if this input layer's class-type is 'layerSkipClass'
```

Comment:
Comment format.
Comment doesn't clarify which layers will be added to `layersToRemove` (if I understand correctly, skipped layers and inputLayer itself will be added).

Reply:
fixed